apps/meteor/app/federation-v2/server/application/RoomServiceSender.ts

```diff
@@ -148,18 +148,50 @@ export class FederationRoomServiceSender {
 		}
 	}
 
-	public async canAddUsersToTheRoom(internalUser: IUser | string, internalRoom: IRoom): Promise<void> {
-		const newUserBeingAdded = typeof internalUser === 'string';
-		if (newUserBeingAdded) {
+	public async canAddUsersToTheRoom(internalUser: IUser | string, internalInviter: IUser, internalRoom: IRoom): Promise<void> {
+		if (!internalRoom.federated) {
 			return;
 		}
+		const tryingToAddNewFederatedUser = typeof internalUser === 'string';
+		if (tryingToAddNewFederatedUser) {
+			throw new Error('error-this-is-an-ee-feature');
+		}
 
-		if (!internalRoom.federated) {
+		const invitee = await this.rocketUserAdapter.getFederatedUserByInternalId((internalUser as IUser)._id);
```

Comment:
the function signature says `internalUser` can be either `IUser | string`, so why are you type casting as `IUser`? should it be typed as `IUser` only on the function signature?

Reply:
Yes, you are right, I forgot to move the validation to the top of the function, but I have already fixed that on my last commit. When the user is a string it should not be possible to add the user.

https://github.com/RocketChat/Rocket.Chat/pull/26283/commits/9200b3d67ea841f6f42042af4324b01fccc30861#diff-818e7b29c0e27fb8420e16fc4c4b9f9ae4cae69e6c4c89f011018da429006edcR155